# HG changeset patch
# User John Donoghue
# Date 1550003512 18000
#      Tue Feb 12 15:31:52 2019 -0500
# Branch stable
# Node ID c66f509d9f5ca52665c2d52157f81ba710b9c6be
# Parent  b3b78bf8311340daa7f10442c188b2fe8f66dd63
Mark qt documentation as readonly (bug #53841).

* libgui/src/documentation.cc: Set qt readonly flag on qt documentation.

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -72,16 +72,21 @@ namespace octave
       collection = QString::fromStdString (config::oct_doc_dir ()
                                            + sys::file_ops::dir_sep_str ()
                                            + "octave_interpreter.qhc");
 
     // Setup the help engine with the original collection, use a writable copy
     // of the original collection and load the help data
     m_help_engine = new QHelpEngine (collection, this);
 
+    // Mark help as readonly to avoid error if collection file is stored in a
+    // readonly location
+    m_help_engine->setProperty ("_q_readonly",
+                                QVariant::fromValue<bool> (true));
+
     QString tmpdir = QDir::tempPath();
     m_collection
       = QString::fromStdString (sys::tempnam (tmpdir.toStdString (),
                                               "oct-qhelp-"));
 
     if (m_help_engine->copyCollectionFile (m_collection))
       m_help_engine->setCollectionFile (m_collection);
     else
